src/services/connection.service.ts

```diff
@@ -0,0 +1,221 @@
+
+    
+    constructor() {
+        this.receiveHandlers = new Map<string, RECIEVE_HANDLER_TYPE>();
+        this.requestsMap = new Map<string, RESPONSE_TYPE>();
+        this.routingTable = new RoutingTable();
+        this.sentMessages = new Map<string, Message>();
+        this.responseDefferedsMap = new Map<string, Deferred<boolean>>();
+    }
+
+    public init(port: string): Promise<void> {
+        return new Promise(async (resolve) => {
+            this.sender = new Sender(port);
+            this.receiver = new Receiver(this, port);
+            this.createRoutingTableCleanupTimer();
+            resolve();
+        });
+    }
+
+    /**
+     * Remove the given message from the sent messages history.
+     *
+     * @param messageGuid - The message hash of the message to remove
+     */
+    public removeSentMessage(senderGuid: string): void {
+        this.sentMessages.delete(senderGuid);
+    }
+
+    /**
+     * Register custom receiver handlers.
+     *
+     * @param messageType - The messageType that the receiver handles
+     * @param implementation - The implementation of the receiver handler
+     */
+    public registerReceiveHandlerForMessageType(messageType: string, implementation: RECIEVE_HANDLER_TYPE): void {
+        this.receiveHandlers.set(messageType, implementation);
+    }
+
+    /**
+     * Send a message to the given destination.
+     *
+     * @param messageType - The message type
+     * @param destination - The destination GUID
+     * @param [body] - The message body
+     */
+    public sendMessage(message: Message, destinationGuid: string, responseImplementation?: RESPONSE_TYPE): Promise<void> {
+        if (destinationGuid) {
+            this.sentMessages.set(destinationGuid, message);
+        }
+        const destinationIp = this.getIpFromRoutingTable(destinationGuid);
+        return this.sendMessageByIp(message, destinationIp, responseImplementation);
+    }
+
+    /**
+     * Send a broadcast to the network.
+     *
+     * @param messageType - The message type
+     * @param [body] - The message body
+     */
+    public sendBroadcast(message: Message, responseImplementation?: RESPONSE_TYPE): Promise<void[]> {
+        const promises = [];
+        for (const guid of this.routingTable.peers.keys()) {
+            promises.push(this.sendMessage(message, guid, responseImplementation));
+        }
+        return Promise.all(promises);
+    }
+
+    /**
+     * Check of messageType of the given message has a known implementation, and executes the implementation.
+     *
+     * @param message - The incoming message
+     * @param senderGuid - The GUID of the sender
+     * @param senderIp - The IP of the sender
+     */
+    public onMessage(message: Message): Promise<void> {
+        return new Promise(async (resolve, reject) => {
+            if (!this.sender) {
+                reject();
+                return;
+            }
+            logger.info(`${message.type} message received`);
```

Comment:
Please consider removing all the logging from this library. I'd expect the consuming project to apply logging rather than our library.